jobs/dynatrace-oneagent/templates/pre-start.erb

```diff
@@ -136,7 +177,16 @@ downloadAgent() {
                 installLog "Dynatrace agent download failed, retrying in $retryTimeout seconds"
             fi
         else
-            installLog "Download was successful"
+            if [[ $validateDownload == true ]]; then
+                validateAgent $2
+                if [[ $? -ne 0 ]]; then
+                    installLog "Download validation failed!"
+                else
+                    break
+                fi
+            else
+                break
+            fi
             break
```

Comment:
Remove this line. This exits the cycle, even if we want to attempt a new download after a failed verification.